### PR TITLE
A11y -  4657 - Unique page titles

### LINKED ISSUE #4657: Unique page <title>s for each webpage
## Background

The individual page `<title>`s are not unique and are all “GC Digital Talent”. Each page should have a unique title so that users can navigate through multiple open tabs.

## Recommendation
Follow this convention for each page `<title>`:

- H1 | GC Digital Talent
- e.g. If the user was on the profile main page = “My Profile | GC Digital Talent”

## Documentation
@MaddyDaigle will be using the [site list](https://go.talent.c4nada.ca/sitemap) to identify all the page titles.

## A11y
Here’s some WCAG documentation on this issue:
Failure of success criterion 2.4.2: Page titled
https://www.w3.org/WAI/WCAG21/Understanding/page-titled.html
Failures
The following are common mistakes that are considered failures of this Success Criterion by the WCAG Working Group.
[F25: Failure of Success Criterion 2.4.2 due to the title of a Web page not identifying the contents](https://www.w3.org/WAI/WCAG21/Techniques/failures/F25)

## Acceptance criteria

- [ ] List of page titles are added to the [site list](https://go.talent.c4nada.ca/sitemap)

### PR DESCRIPTION
**Pull request by esizer**
## 👋 Introduction

This adds an `<SEO />` component to control page meta data and implements it on each page to apply unique titles.

## ✅  TO DO

- [x] Get missing titles
- [x] Send off missing translations
- [x] Apply provided translations

### CMO Assets

Opted to not do the CMO Assets since they are being removed, #4991 

## 🧪 Testing

1. Build all projects `npm run production --workspaces`
2. Navigate to pages and ensure the title (in your browser tab) updates accordingly

## 📑 Page Titles

| URL                                                                    | Page <title> EN                 | Page <title> FR                    | Completed? |
| ---------------------------------------------------------------------- | ------------------------------- | ---------------------------------- | ---------- |
| /en                                                                    | Welcome                         | Bienvenue                          | ✅          |
| /en/browse/pools                                                       | Browse opportunities            | Parcourez les opportunités         | ✅          |
| /en/pools/{:poolId}                                                    | \[Pool Name\]              | \[Pool Name\]                      | ✅          |
| /en/applications/{:applicationId}/apply                                | Application Profile             | Profil de candidature              | ✅          |
| /en/applications/{:applicationId}/submit                               | Submit Application Profile      | Soumettre un profil de demande                                 | ✅          |
| /en/search                                                             | Search Pools                    | Researche bassin de talents        | ✅          |
| /en/search/request                                                     | Request Pool Candidates       | Demander des candidats faisant partie d’un bassin  | ✅          |
| /en/login-info                                                         | Login using GC Key              | Connexion à l'aide de la CléGC     | ✅          |
| /en/register                                                           | Register                        | Enregistrer                        | ✅          |
| /en/create-account                                                     | Create account                  | Créer un compte                                 | ✅          |
| /en/users/{:userId}/profile                                            | My Profile                      | Mon profil                         | ✅          |
| /en/users/{:userId}/profile/about-me/edit                              | About Me                        | À propos de moi                    | ✅          |
| /en/users/{:userId}/profile/language-info/edit                         | Language Information            | Informations linguistiques         | ✅          |
| /en/users/{:userId}/profile/government-info/edit                       | Government Information          | Renseignements du gouvernement     | ✅          |
| /en/users/{:userId}/profile/work-location/edit                         | Work location                   | Lieu de travail                    | ✅          |
| /en/users/{:userId}/profile/work-preferences/edit                      | Work preferences                | Préférences de travail             | ✅          |
| /en/users/{:userId}/profile/role-salary-expectations/edit              | Role and Salary Expectations    | Rôle et salaire escomptés          | ✅          |
| /en/users/{:userId}/profile/employment-equity/edit                     | Diversity, equity and inclusion | Diversité, équité et inclusion     | ✅          |
| /en/users/{:userId}/profile/experiences/                               | My experience and skills        | Mon expérience et mes compétences  | ✅          |
| /en/users/{:userId}/profile/experiences/awards/create                  | Edit Award                      | Modifier le prix                   | ✅          |
| /en/users/{:userId}/profile/experiences/awards/{:experienceId}/edit    | Edit award                      | Modifier le prix                   | ✅          |
| /en/users/{:userId}/profile/experiences/education/create               | Edit education experience       | Modifier l’expérience de scolarité | ✅          |
| /en/users/{:userId}/profile/experiences/education/{:experienceId}/edit | Edit education experience       | Modifier l’expérience de scolarité | ✅          |
| /en/users/{:userId}/profile/experiences/personal/create                | Edit personal experience        | Modifier l’expérience personnelle  | ✅          |
| /en/users/{:userId}/profile/experiences/personal/{:experienceId}/edit  | Edit personal experience        | Modifier l’expérience personnelle  | ✅          |
| /en/users/{:userId}/profile/experiences/work/create                    | Edit work experience            | Modifier l’expérience de travail   | ✅          |
| /en/users/{:userId}/profile/experiences/work/{:experienceId}/edit      | Edit work experience            | Modifier l’expérience de travail   | ✅          |
| /en/users/{:userId}/profile/experiences/community/create               | Edit community experience       | Modifier expérience - collectivité | ✅          |
| /en/users/{:userId}/profile/experiences/community/{:experienceId}/edit | Edit community experience       | Modifier expérience - collectivité | ✅          |
| /en/users/{:userId}/applications                                       | My applications                 | Mes candidatures                   | ✅          |
| /en/admin/dashboard                                                    | Admin - GC Digital Talent       | Admin - Talent numérique du GC     | ✅          |
| /en/admin/pools                                                        | Pools                           | Bassins                            | ✅          |
| /en/admin/pools/create                                                 | Create Pool                     | Créer un bassin                    | ✅          |
| /en/admin/pools/{:poolId}                                              | View Pool                       | Annonce du bassin                  | ✅          |
| /en/admin/pools/{:poolId}/edit                                         | Edit Pool                       | Modifier - bassin                  | ✅          |
| /en/admin/users                                                        | Users                           | Utilisateurs                       | ✅          |
| /en/admin/users/{:userId}                                              | Candidate Details               | Détails sur le candidat            | ✅          |
| /en/admin/users/{:userId}/edit                                         | Update User                     | Mettre à jour l’utilisateur        | ✅          |
| /en/admin/talent-requests                                              | Talent Requests                 | Demandes de talent                 | ✅          |
| /en/admin/talent-requests/{:id}                                        | View Request                    | Afficher la demande                | ✅          |
| /en/admin/settings/classifications                                     | Classifications                 | Classifications                    | ✅          |
| /en/admin/settings/classifications/create                              | Create Classification           | Créer une classification           | ✅          |
| /en/admin/settings/classifications/{:classificationId}/edit            | Update Classification           | Modifier - classification          | ✅          |
| /en/admin/settings/skills                                              | Skills                          | Compétences                        | ✅          |
| /en/admin/settings/skills/create                                       | Create Skill                    | Créer une compétence               | ✅          |
| /en/admin/settings/skills/{:skillId}/edit                              | Edit Skill                      | Modifier - compétence              | ✅          |
| /en/admin/settings/skills/families                                     | Skill Families                  | Familles de compétence             | ✅          |
| /en/admin/settings/skills/families/create                              | Create Skill Family             | Créer - famille                    | ✅          |
| /en/admin/settings/skills/families/{:skillFamilyId}/edit               | Edit Skill Family               | Modifier - famille                 | ✅          |
| /en/admin/settings/departments                                         | Departments                     | Ministères                         | ✅          |
| /en/admin/settings/departments/create                                  | Create Department               | Créer un ministère                 | ✅          |
| /en/admin/settings/departments/{:departmentId}/edit                    | Edit Department                 | Mettre à jour un ministère         | ✅          |
| /en/admin/cmo-assets                                                   | CMO Assets                      | Actifs du BGC                      | ⛔️         |
| /en/admin/cmo-assets/create                                            | Create CMO Asset                | Créer un actif du BGC              | ⛔️         |
| /en/admin/cmo-assets/{:id}/edit                                        | Edit CMO Asset                  | Modifier - actif du BGC            | ⛔️         |

## 🤖 Robot stuff

Resolves #4657

### CODE REVIEW

File: frontend/admin/src/js/components/classification/ClassificationPage.tsx [hunk 1]
```diff
@@ -11,17 +12,26 @@ export const ClassificationPage: React.FC = () => {
...
-    </DashboardContentContainer>
+      />
+      <DashboardContentContainer>
+        <PageHeader icon={TagIcon}>
+          {intl.formatMessage({
+            defaultMessage: "Classifications",
+            id: "hnyrdl",
+            description:
+              "Heading displayed above the Classification Table component.",
+          })}
```
Comment (yonikid15) [inline frontend/admin/src/js/components/classification/ClassificationPage.tsx:30]: Can the duplicate strings be turned into `pageTitle` variable too?

File: frontend/admin/src/js/components/pool/PoolPage.tsx [hunk 2]
```diff
@@ -3,24 +3,33 @@ import { useIntl } from "react-intl";
...
-      <PoolTableApi />
-    </DashboardContentContainer>
+      />
+      <DashboardContentContainer>
+        <PageHeader icon={Squares2X2Icon}>
+          {intl.formatMessage({
+            defaultMessage: "Pools",
+            id: "qL43Gw",
+            description: "Heading displayed above the Pool Table component.",
+          })}
```
Comment (yonikid15) [inline frontend/admin/src/js/components/pool/PoolPage.tsx:28]: Can the duplicate strings be turned into `pageTitle` variable too?

File: frontend/admin/src/js/components/skillFamily/SkillFamilyPage.tsx [hunk 3]
```diff
@@ -11,17 +12,26 @@ export const SkillFamilyPage: React.FC = () => {
...
+          {intl.formatMessage({
+            defaultMessage: "Skill Families",
+            id: "w73KdO",
+            description:
+              "Heading displayed above the Skill Family Table component.",
+          })}
+        </PageHeader>
+        <SkillFamilyTableApi />
+      </DashboardContentContainer>
+    </>
```
Comment (yonikid15) [inline frontend/admin/src/js/components/skillFamily/SkillFamilyPage.tsx:34]: Same thing here 😄

File: frontend/admin/src/js/components/user/UserPage.tsx [hunk 4]
```diff
@@ -9,25 +11,34 @@ import DashboardContentContainer from "../DashboardContentContainer";
...
-    </DashboardContentContainer>
+      />
+      <DashboardContentContainer>
+        <PageHeader icon={UserIcon}>
+          {intl.formatMessage({
+            defaultMessage: "Users",
+            id: "8c/kP1",
+            description: "Title for users page on the admin portal.",
+          })}
+        </PageHeader>
```
Comment (yonikid15) [inline frontend/admin/src/js/components/user/UserPage.tsx:29]: Same thing here 😄

**Review by yonikid15 — CHANGES_REQUESTED** [review]
Excellent work Eric 👍 Everything is working as expected. I just requested some minor changes for some duplicate messages. After that should be good to go.

File: frontend/admin/src/js/components/classification/ClassificationPage.tsx (hunk 1 above)
Comment (esizer) [reply frontend/admin/src/js/components/classification/ClassificationPage.tsx]: Goo catch! You should see there are all resolved in [reduce duplicate intl strings](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5024/commits/a60a6630586bfb6ab864c59411cb5602500ef970)

**Review by yonikid15 — APPROVED** [review]
Awesome 🥇

**Pull request merged**